x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/date/DateDiff.java

```diff
@@ -164,7 +164,9 @@ public static Part resolve(String dateTimeUnit) {
         examples = { @Example(file = "date", tag = "docsDateDiff"), @Example(description = """
             When subtracting in calendar units - like year, month a.s.o. - only the fully elapsed units are counted.
             To avoid this and obtain also remainders, simply switch to the next smaller unit and do the date math accordingly.
-            """, file = "date", tag = "evalDateDiffYearForDocs") }
+            """, file = "date", tag = "evalDateDiffYearForDocs"), @Example(description = """
+            If any column is multivalued, this will return a `null` result.
```

Comment:
We'd been considering this behavior to be "undefined" but the `null-and-warning` behavior as something we may change in the future. I'm not sure that that's really true though - folks may be relying on it even though it's a warning.